src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -0,0 +1,97 @@
+import { Frame, PauseId, TimeStampedPoint } from "@replayio/protocol";
+import cloneDeep from "lodash/cloneDeep";
+import { createCache } from "suspense";
+
+import { framesCache } from "replay-next/src/suspense/FrameCache";
+import { objectCache } from "replay-next/src/suspense/ObjectPreviews";
+import { pauseEvaluationsCache, pauseIdCache } from "replay-next/src/suspense/PauseCache";
+import { ReplayClientInterface } from "shared/client/types";
+import { UserActionEvent } from "shared/test-suites/types";
+
+export type Type = {
+  count: number | null;
+  pauseId: PauseId;
+  props: Object | null;
+};
+
+export const TestEventDetailsCache = createCache<
+  [client: ReplayClientInterface, timeStampedPoint: TimeStampedPoint, variable: string],
+  Type
+>({
+  debugLabel: "TestEventDetailsCache",
+  getKey: ([client, timeStampedPoint, variable]) => `${timeStampedPoint}:${variable}`,
+  load: async ([client, timeStampedPoint, variable]) => {
+    const endPauseId = await pauseIdCache.readAsync(
+      client,
+      timeStampedPoint.point,
+      timeStampedPoint.time
+    );
+    let frames: Frame[] | undefined;
+    try {
+      frames = await framesCache.readAsync(client, endPauseId);
+    } catch (error) {
+      // TODO [FE-1555] RUN currently throws in some cases;
+      // degrade gracefully in this case by just disabling the details panel
+      console.error(error);
+    }
+
+    const callerFrameId = frames?.[1]?.frameId;
```

Comment:
:hand: would be nice to have a comment explaining _why_ we're looking at `[1]`.  I assume we're trying to go back before Cypress or something?

Reply:
I didn't write this. I just ported it over from the less usable format here:

https://github.com/replayio/devtools/blob/2406ee28d7aeca034dad142bfed72852d047f183/src/ui/components/TestSuite/suspense/consoleProps.ts#L22-L40